plugins/out_s3/s3.c

```diff
@@ -657,9 +688,143 @@ static int cb_s3_init(struct flb_output_instance *ins,
+            return -1;
+        }
+        flb_pclose(cmdp);
+
+        tmp = flb_output_get_property("parquet.compression", ins);
+        if (tmp != NULL) {
+            if (strncasecmp(tmp, "uncompressed", 12) == 0 ||
+                strncasecmp(tmp, "snappy", 6) == 0 ||
+                strncasecmp(tmp, "gzip", 4) == 0 ||
+                strncasecmp(tmp, "zstd", 4) == 0) {
+                flb_plg_info(ctx->ins, "parquet.compression format is %s", tmp);
+            }
+            else if (strncasecmp(tmp, "lzo", 3) == 0 ||
+                strncasecmp(tmp, "brotli", 6) == 0 ||
+                strncasecmp(tmp, "lz4", 3) == 0) {
+                flb_plg_info(ctx->ins, "unsupported parquet.compression format %s", tmp);
+            }
+            else {
+                flb_plg_error(ctx->ins, "unknown parquet.compression format %s", tmp);
+                return -1;
+            }
+            for (i = 0; i < strlen(tmp); i++) {
+                buf[i] = toupper(tmp[i]);
+            }
+
+            ctx->parquet_compression = flb_sds_create_len(buf, strlen(buf));
+            if (ctx->parquet_compression == NULL) {
+                flb_plg_error(ctx->ins, "Failed to create parquet compression type");
+                flb_errno();
+                return -1;
+            }
+        }
+        else {
+            ctx->parquet_compression = \
+                    flb_sds_create(DEFAULT_PARQUET_COMPRESSION_FORMAT_UPCASES);
+            if (ctx->parquet_compression == NULL) {
+                flb_plg_error(ctx->ins, "Failed to create parquet compression type");
+                flb_errno();
+                return -1;
+            }
+            flb_plg_debug(ctx->ins, "parquet.compression format is %s",
+                          DEFAULT_PARQUET_COMPRESSION_FORMAT_UPCASES);
+        }
+
+        tmp = flb_output_get_property("parquet.record_type", ins);
+        if (!tmp) {
+            flb_plg_info(ctx->ins, "parquet.record_type format is %s",
+                         DEFAULT_PARQUET_RECORD_TYPE);
+            ctx->parquet_record_type = \
+                    flb_sds_create(DEFAULT_PARQUET_RECORD_TYPE);
+            if (ctx->parquet_record_type == NULL) {
+                flb_plg_error(ctx->ins, "Failed to create parquet record type");
+                flb_errno();
+                return -1;
+            }
+        }
+        else {
+            if (strncasecmp(tmp, "json", 4) == 0) {
+                tmp = "jsonl"; /* json should be interpreted as jsonl */
+                flb_plg_info(ctx->ins, "parquet.record_type format is %s", tmp);
+            }
+            else if (strncasecmp(tmp, "msgpack", 7) == 0 ||
+                strncasecmp(tmp, "jsonl", 5) == 0) {
+                flb_plg_info(ctx->ins, "parquet.record_type format is %s", tmp);
+            }
+            else {
+                flb_plg_error(ctx->ins, "unknown parquet.record_type format %s", tmp);
```

Comment:
As we are exiting control here, I am not seeing cleanup of:
- ctx->parquet_compression
- ctx->parquet_record_type